app/lib/pre_assembly/assembly_directory.rb

```diff
@@ -0,0 +1,55 @@
+module PreAssembly
+  # Represents the assembly structure on the filesystem
+  class AssemblyDirectory
+    def self.create(druid_id:)
+      new(druid_id: druid_id).tap(&:create_object_directories)
```

Comment:
Just to be explicit: the use of `#tap` here accomplishes the same effect that could also be achieved by calling `new(druid_id: druid_id).create_object_directories` if `#create_object_directories` returned `self`? Using `#tap` keeps the extracted code closer to the original code, so :+1:.